src/openassetio-core/include/openassetio/specifications.hpp

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2022 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <variant>
+#include <vector>
+// API export generated header
+#include <openassetio/export.h>
+
```

Comment:
[fix] Add High-level architectural manifesto of what is in this file before too much implementation happens 📜

Reply:
Many things changed, but somewhat related, added some docstrings in 6a50c5f9

Reply:
These are a good start. I guess the main request was for something more along the lines of what even is a spec, when is used, the whole view/data separation and relation to traits.

Reply:
Did a second pass in 55fbc07a. Also enabled Doxygen for public C++ headers in c905c47d.

Perhaps still need a little fleshing out - more example snippets, perhaps.

That reminds me, one cool thing I've done with Doxygen in the long lost past is [snippets](https://www.doxygen.nl/manual/commands.html#cmdsnippet), where you can mark a section of source code with special comments, then refer to that section in the docs, which will reproduce it.  So you have a living example.  Would be great to eventually have some "workflow test" snippets that we can litter through the docs.

Reply:
Restructured a bit in d5a7251b - how's it looking re. this review comment?

Reply:
That is spot on!